src/commands/UserUpsertCommand.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  MultisigUserUpdateMessage,
+  USER_UPDATE_MESSAGE_SCHEMA,
+} from "../lib/users/schemas.js";
+import { isTypedMessage } from "../utils/signatures.js";
+import UserUpsertSignatureVerifier from "../lib/safe-signature-verification/UserUpsertSignatureVerifier.js";
+import { Database } from "../types/supabaseData.js";
+
+import { SafeApiCommand } from "./SafeApiCommand.js";
+
+type SignatureRequest =
+  Database["public"]["Tables"]["signature_requests"]["Row"];
+
+export class UserUpsertCommand extends SafeApiCommand {
+  async execute(): Promise<void> {
+    const signatureRequest = await this.dataService.getSignatureRequest(
+      this.safeAddress,
+      this.messageHash,
+    );
+
+    if (!signatureRequest || signatureRequest.status !== "pending") {
+      return;
```

Comment:
If it isn't pending, should a warning or error be logged?

Reply:
I don't think it's a case that we should warn about. It just means that the developer who changed the query code after me made a mistake and included non-pending requests in their query, but nonetheless they should just be ignored. We could print something, just to make us aware that this mistake was made, of course. WDYT?